src/pyscipopt/scip.pyx

```diff
@@ -679,12 +691,29 @@ cdef class Model:
     def __dealloc__(self):
         # call C function directly, because we can no longer call this object's methods, according to
         # http://docs.cython.org/src/reference/extension_types.html#finalization-dealloc
-        PY_SCIP_CALL( SCIPfree(&self._scip) )
+        if self._scip is not NULL and self.freescip:
+           PY_SCIP_CALL( SCIPfree(&self._scip) )
 
     def create(self):
         """Create a new SCIP instance"""
         PY_SCIP_CALL(SCIPcreate(&self._scip))
 
+    @staticmethod
+    cdef createModel(SCIP* scip):
+        """Creates a model and appropriately assigns the scip and bestsol parameters
+        """
+        if scip == NULL:
+            raise Warning("cannot create Model with SCIP* == NULL")
+        model = Model(noscipcreate=True)
+        model._scip = scip
+        model._bestSol = Solution.create(SCIPgetBestSol(scip))
+        return model
+
+    def createModelFromSCIP(self):
```

Comment:
Why not call it `copy(self)`?

Reply:
As I mentioned above, this could be confused with the copy methods of the C-API. Suggestion above.